core/identity-spi/src/main/java/software/amazon/awssdk/identity/spi/internal/ProviderSource.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.identity.spi.internal;
+
+import java.util.Optional;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.annotations.ThreadSafe;
+
+/**
+ * Interface to represent the provider that is the source of this identity.
+ */
+@SdkInternalApi
+@ThreadSafe
+public interface ProviderSource {
```

Comment:
1. This cannot be internal, since it's implemented by public classes.
2. The name is weird. It sounds like it provides providers? Maybe "Provided" or "ProviderAware"?

Reply:
updated